sdk/servicebus/microsoft-azure-servicebus/src/test/java/com/microsoft/azure/servicebus/jms/SBJmsConnectionFactory.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.azure.servicebus.jms;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSContext;
+import javax.jms.JMSException;
+import org.apache.qpid.jms.JmsConnectionFactory;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+
+public class SBJmsConnectionFactory implements ConnectionFactory {
+    private JmsConnectionFactory factory;
+
+    private SBJmsConnectionFactory(String sasKeyName, String sasKey, String host) {
+        if (sasKeyName == null || sasKeyName == null || host == null) {
+            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
+        }
+        
+        this.factory = new JmsConnectionFactory(sasKeyName, sasKey, "amqps://" + host + "?amqp.idleTimeout=120000&amqp.traceFrames=true");
```

Comment:
I don't think we can have `traceFrames=true` everytime. We may need to expose that as a parameter.

Reply:
I copy pasted that off our doc online, don't see any restrictions on it...

Reply:
+1 we should expose these as properties to allow people to choose.